### ISSUE TITLE
InnerTypeLastCheck: false negative on compact source files

### ISSUE THREAD

**Issue by ZaheerAhmadDev** [checkstyle/checkstyle#20700]
Parent Issue #20590

I have read check documentation: https://checkstyle.org/checks/design/innertypelast.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="InnerTypeLast"/>
  </module>
</module>
```

```java
$ cat src/InputInnerTypeLastCompact.java
class Inner { }

void main() { }
```

```
$ java  -jar checkstyle-13.8.0-all.jar -c config.xml src/InputInnerTypeLastCompact.java
Starting audit...
Audit done.
```

**Describe what you expect in detail.**
No violation is reported but per the check own documented example a member
declared after an inner type should violate

```java
$ cat src/Test.java
class Test1 {
  private String s;
  class InnerTest1 {}
  public void test() {} // violation
}

$ java  -jar checkstyle-13.8.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] C:\Users\PMYLS\Desktop\check\src\Test.java:4:3: Init blocks, constructors, fields and methods should be before inner types. [InnerTypeLast]
Audit done.
Checkstyle ends with 1 errors.

```

This is confirmed by wrapping the same code in an ordinary named class which
correctly reports the violation

```java
$ cat src/InputInnerTypeLastCompact.java
public class InputInnerTypeLastCompact {
    class Inner { }

    void main() { }
}
```

```
$ java  -jar checkstyle-13.8.0-all.jar -c config.xml src/InputInnerTypeLastCompact.java
Starting audit...
[ERROR] C:\Users\PMYLS\Desktop\check\src\InputInnerTypeLastCompact.java:4:5: Init blocks, constructors, fields and methods should be before inner types. [InnerTypeLast]
Audit done.
Checkstyle ends with 1 errors.
```

**Side effect of fix:**
now content of Enums are also validated.

**Comment by ZaheerAhmadDev** [comment]
I am on it.

**Comment by romani** [comment]
fix is merged
